features/0211-route-coordination/README.md

```diff
@@ -0,0 +1,144 @@
+# 0211: Route Coordination Protocol
+- Authors: [Sam Curren](telegramsam@gmail.com)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2019-09-03
+- Status Note: Initial version 
+- Start Date: 2019-09-03
+- Tags: feature, protocol
+
+## Summary
+
+A protocol to coordinate routing configuration between a routing agent and the recipient.
+
+## Motivation
+
+Use of the forward message in the Routing Protocol requires an exchange of information. The Recipient must know which endpoint and routing key(s) to share, and the Router needs to know which keys should be routed via this relationship.
+
+## Tutorial
+
+### Roles
+
+**router** - The agent that will be receiving `forward` messages on behalf of the _recipient_.
+**recipient** - The agent for whom the `forward` message payload is intended.
+
+### Flow
+A recipient may discover an agent capable of routing using the Feature Discovery Protocol. If protocol is supported with the _router_ role, a _recipient_ may send a `route_request` to initiate a routing relationship.
+
+First, the _recipient_ sends a `route_request` message to the _router_. If the _router_ is willing to route messages, it will respond with a `route_grant` message. The _recipient_ will share the routing information in the grant message with other contacts.
+
+When a new key is used by the _recipient_, it must be registered with the _router_ to enable route identitifcation. This is done with a `keylist_update` message.
+
+The `keylist_update` and `keylist_query` methods are used over time to identify and remove keys that are no longer in use by the _recipient_.
+
+
+
+## Reference
+
+### Route Request
+This message serves as a request from the recipient to the router, asking for the permission (and routing information) to publish the endpoint as a router.
+```jsonc
+{
+    "@id": "123456781",
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routecoordination/1.0/route-request"
+}
+```
+### Route Grant
+A route grant message is a signal from the router to the recipient that permission is given to distribute the included information as an inbound route.
+```jsonc
+{
+    "@id": "123456781",
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routecoordination/1.0/route-grant",
+    "endpoint": "",
+    "routing_keys": []
+    
+}
+```
+Questions:
+- What about multiple endpoint options? http and ws?
+
+### Keylist Update
+Used to notify the _router_ of keys in use by the _recipient_.
+```jsonc
+{
+    "@id": "123456781",
+    "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routecoordination/1.0/keylist_update",
+    "updates":[
+        {
+            "recipient_key": "",
```

Comment:
I do not understand why the Routing Agent needs to be aware of the recipient's keys. The RA should identify the recipient of the encrypted msg using the `to` field of the `Forward` msg: https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0094-cross-domain-messaging#required-end-to-end-encryption-of-the-agent-message. Furthermore, the `Forward` message does not include these keys.

I would presume the DID referenced in `to` would be shared by the recipient with the RA and the Sender...